src/ucp/wireup/address.c

```diff
@@ -517,29 +555,37 @@ static ucs_status_t ucp_address_do_pack(ucp_worker_h worker, ucp_ep_h ep,
 
             /* Transport name checksum */
             *(uint16_t*)ptr = context->tl_rscs[i].tl_name_csum;
-            ptr += sizeof(uint16_t);
+            ptr = UCS_PTR_TYPE_OFFSET(ptr, context->tl_rscs[i].tl_name_csum);
 
             /* Transport information */
             attr_len = ucp_address_pack_iface_attr(worker, ptr, i, iface_attr,
                                                    worker->atomic_tls & UCS_BIT(i));
             ucp_address_memchek(ptr, attr_len,
                                 &context->tl_rscs[dev->rsc_index].tl_rsc);
 
-            iface_addr_len = iface_attr->iface_addr_len;
-            flags_ptr      = ucp_address_iface_flags_ptr(worker, ptr, attr_len);
-            ptr           += attr_len;
+            if (flags & UCP_ADDRESS_PACK_FLAG_IFACE_ADDR) {
+                iface_addr_len = iface_attr->iface_addr_len;
+            } else {
+                iface_addr_len = 0;
+            }
+            flags_ptr = ucp_address_iface_flags_ptr(worker, ptr, attr_len);
+            ptr       = UCS_PTR_BYTE_OFFSET(ptr, attr_len);
             ucs_assertv(iface_addr_len < UCP_ADDRESS_FLAG_EP_ADDR,
                         "iface_addr_len=%zu", iface_addr_len);
 
             /* Pack iface address */
-            ptr    = ucp_address_pack_length(worker, ptr, iface_addr_len);
-            status = uct_iface_get_address(wiface->iface, (uct_iface_addr_t*)ptr);
-            if (status != UCS_OK) {
-                return status;
+            ptr = ucp_address_pack_length(worker, ptr, iface_addr_len);
```

Comment:
are we packing iface address length, but not iface address itself?

Reply:
just to simplify unpacking, address format is non-trivial to do everything by flags, some stuff easier to send than handle.